docs/scaling/multi-node-setup.rst

```diff
@@ -4,162 +4,237 @@
-way to prevent other nodes from joining a cluster is to give it a different and
-unique name.
 
-.. code-block:: yaml
+.. _auto-bootstrapping:
 
-    cluster.name: my_cluster
+Single host auto-bootstrapping
+------------------------------
 
-Node name
----------
+If you start up several nodes (with default configuration) on a single host,
+they will automatically discover one another and form a cluster.
+
+If you want to run CrateDB on your local machine for development or
+experimentation purposes, this is probably your best bootstrapping option.
+
+.. CAUTION::
+
+    Single host auto-bootstrapping is useful for development environments.
+    However, auto-bootstrapping is typically too slow for :ref:`production use
+    <going-into-production>`. Additionally, for improved performance and
+    resiliency, production CrateDB clusters should be run with :ref:`one node
+    per host machine <manual-bootstrapping>`. If you use multiple hosts, you
+    must configure your cluster with :ref:`manual bootstrapping
+    <manual-bootstrapping>`.
+
+.. WARNING::
+
+    If you start multiple nodes on different hosts with auto-bootstrapping
+    enabled, you cannot, at a later point, bring those nodes together for form
+    a single cluster--there is no way to merge CrateDB clusters without the
+    risk of data loss.
+
+    If you have multiple nodes running on different hosts, you can check
+    whether they have formed independent clusters by visiting the `Admin UI`_
+    (which runs on every node) and checking the `cluster browser`_.
+
+    If you find yourself with multiple independent clusters and you instead
+    want to form a single cluster, follow these steps:
+
+      1. `Back up your data`_
+      2. Shut down all the nodes
+      3. Completely wipe each node by deleting the contents of the ``data``
+         directory under `CRATE_HOME`_
+      4. Follow in instructions in the next section (:ref:`manual bootstrapping
+         <manual-bootstrapping>`)
+      5. Restart all the nodes and verify that they have formed a single cluster
+      6. Restore your data
+
+
+.. _auto-bootstrapping-tarball:
+
+Tarball install
+~~~~~~~~~~~~~~~
+
+If you're installing CrateDB using the `tarball method`_, you can start a
+single host three-node cluster with auto-bootstrapping by following these
+instructions.
+
+Unpack the tarball:
+
+.. code-block:: console
+
+    sh$ tar -xzf crate-*.tar.gz
+
+Copy the expanded directory three times, one for each node:
+
+.. code-block:: console
+
+    sh$ cp -R crate-*/ node1
+    sh$ cp -R crate-*/ node2
+    sh$ cp -R crate-*/ node3
+
+.. TIP::
+
+    Each directory will function as `CRATE_HOME`_ for that node
+
+Because you want to run a multi-node cluster, you must configure the metadata
+gateway so that CrateDB knows how to recover its state. For a three-node cluster,
+set `gateway-expected-nodes`_ to ``3`` and set `gateway-recover-after-nodes`_
+to ``3``.
```

Comment:
For the development auto-bootstrapping process this is not strictly necessary. It's more of a resiliency /safe-guard for production environments, to make sure a subset of the nodes don't recover a slightly outdated state without waiting that all nodes have been discovered.

Reply:
how about this:

> Configuring the metadata gateway is a safeguarding mechanism that is useful for production clusters. It is not strictly necessary when running in development. However, the Admin UI will issue warnings if the metadata gateway is not configured.

I added this step here because I don't want a reader's first experience with a CrateDB cluster to involve an Admin UI issuing warnings. that could alarm/concern some people

pushing a fixup